### PR TITLE
fix issue that causes MissingPreviousEpoch error in CI

### LINKED ISSUE #860: Investigate `MissingPreviousEpoch` error in CI
See https://github.com/fedimint/fedimint/actions/runs/3352115861/jobs/5554034294#step:9:2806

### PR DESCRIPTION
**Pull request by jkitman**
There was a lot of flakiness in the reconnect tests that this should address, while also making them faster.  Also sets a max delay on client requests to 1sec.

Fixes #860 or at least improves it...if anyone encounters again please ping me.

### CODE REVIEW

File: scripts/reconnect-test.sh [hunk 1]
```diff
@@ -12,34 +12,33 @@ server3=$(tail -3 $FM_PID_FILE | head -1)
...
-kill $server1
-mine_blocks 100
-await_block_sync
+kill_server $server1
 mine_blocks 100
 await_block_sync
+
 ./scripts/start-fed.sh
+await_all_peers
+echo "Successfully rejoined!!!"
```
Comment (dpc) [inline scripts/reconnect-test.sh:33]: Seems a bit ... casual and not informative. :D ."Mint 1 successfully rejoined federation"?

File: scripts/reconnect-test.sh [hunk 2]
```diff
@@ -12,34 +12,33 @@ server3=$(tail -3 $FM_PID_FILE | head -1)
...
-echo "Kill server3..."
-kill $server3
-echo "Kill server4..."
-kill $server4
-sleep 15
 ./scripts/start-fed.sh
-mine_blocks 100
-await_block_sync
+await_all_peers
+echo "Successfully restarted!!!"
```
Comment (dpc) [inline scripts/reconnect-test.sh:44]: "Successfully restarted the Federation consensus"?

File: scripts/reconnect-test.sh [hunk 3]
```diff
@@ -12,34 +12,33 @@ server3=$(tail -3 $FM_PID_FILE | head -1)
 server2=$(tail -2 $FM_PID_FILE | head -1)
 server1=$(tail -1 $FM_PID_FILE | head -1)
 
+function kill_server() {
+  echo "Killing server $1..."
+  kill $1
+  tail --pid=$1 -f /dev/null
```
Comment (dpc) [inline scripts/reconnect-test.sh:18]: Oh, I didn't know this one! :+1:

File: scripts/lib.sh [hunk 4]
```diff
@@ -27,6 +27,10 @@ function await_block_sync() {
   echo "Mint at ${EXPECTED_BLOCK_HEIGHT}H"
 }
 
+function await_all_peers() {
```
Comment (dpc) [inline scripts/lib.sh:30]: Does this await *all* peers or *enough to form a consensus* clients?

Comment (jkitman) [reply scripts/lib.sh:30]: The new `api` command in the CLI waits for all peers, which allowed me to refactor this script

File: fedimint-server/src/lib.rs [hunk 5]
```diff
@@ -253,8 +253,10 @@ impl FedimintServer {
                 EpochHistory::new(last_outcome.epoch, contributions, &prev_epoch)
             } else {
                 let epoch_pk = self.cfg.epoch_pk_set.public_key();
-                let result = self.api.fetch_epoch_history(epoch_num, epoch_pk).await;
-                result.map_err(|_| EpochVerifyError::MissingPreviousEpoch)?
+                self.api
+                    .fetch_epoch_history(epoch_num, epoch_pk)
+                    .await
+                    .expect("fetches history")
```
[the remark this answers is not in the document]
Comment (jkitman) [reply fedimint-server/src/lib.rs:259]: The reason for that is the error was getting swallowed and if this fails we just panic immediately anyway.

I'd like to see the error message if we hit this again so I have a better idea how we should handle it.

**Review by elsirion — APPROVED** [review]

**Pull request merged**